lib/broadway_rabbitmq/channel_pool.ex

```diff
@@ -0,0 +1,20 @@
+defmodule BroadwayRabbitMQ.ChannelPool do
+  @moduledoc """
+  A behaviour used to handle channel pooling to a custom pool.
+  If you want to use a pool with BroadwayRabbitMQ, your pool must implement this behaviour.
+  When configuring the BroadwayRabbitMQ connection, set the value `{:custom_pool, module, options}`.
+  Where module is any module implementing this behaviour and options will be passed to the callbacks.
+  """
+
+  @doc """
+  Invoked when BroadwayRabbitMQ needs a channel.
+  """
+  @callback checkout_channel(options :: term()) ::
+              {:ok, AMQP.Channel.t()} | {:error, reason :: term()}
```

Comment:
My suggestion is to make this return:

```suggestion
  @callback checkout_channel(args :: term()) ::
              {:ok, AMQP.Channel.t()} | {:error, reason :: Exception.t()}
```

so that we can format it with `Exception.message/1` and get nicer errors. Same for `checkin_channel/1`.